### PR TITLE
[ui-library-add-call-recording] Convert backend to .NET Azure Functions

### PR DESCRIPTION
**Pull request by prprabhu-ms**
## Purpose

Call automation SDKs are only available in Java and C#.

## How to Test

See `./ui-library-add-call-recording/README.md` for instructions for running locally.

### CODE REVIEW

File: ui-library-add-call-recording/api/host.json [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
-      }
-    }
-  },
-  "extensionBundle": {
-    "id": "Microsoft.Azure.Functions.ExtensionBundle",
-    "version": "[2.*, 3.0.0)"
-  }
-}
+{
+    "version": "2.0",
```
Comment (alkwa-msft) [inline ui-library-add-call-recording/api/host.json:2]: i am curious if we want to mention applicationInsights here. Should we just delete it to be cleaner?

File: ui-library-add-call-recording/api/stopRecording.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Contoso
+{
+    public static class stopRecording
```
Comment (alkwa-msft) [inline ui-library-add-call-recording/api/stopRecording.cs:13]: Classes start with uppercase

image


https://www.c-sharpcorner.com/UploadFile/8a67c0/C-Sharp-coding-standards-and-naming-conventions/#:~:text=C%23%20Naming%20Conventions%20%20%20Abbreviations%20%20,%20%20DropDownList%20%2021%20more%20rows%20

File: ui-library-add-call-recording/api/types.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace Contoso
+{
+    class Result
+    {
+        public string text { get; set; }
```
Comment (alkwa-msft) [inline ui-library-add-call-recording/api/types.cs:5]: properties should be UpperCamelCase

File: ui-library-add-call-recording/api/startRecording.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+namespace Contoso
+{
+    public static class startRecording
+    {
+        [FunctionName("startRecording")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
```
Comment (alkwa-msft) [inline ui-library-add-call-recording/api/startRecording.cs:20]: I am curious if the request body has nothing and we try to read it. Do we need to do some try/catch magic here?

**Review by alkwa-msft — APPROVED** [review]

File: ui-library-add-call-recording/api/host.json (hunk 1 above)
Comment (prprabhu-ms) [reply ui-library-add-call-recording/api/host.json:2]: This is the default template generated by vscode for azure functions. Will look at removing this.

File: ui-library-add-call-recording/api/stopRecording.cs (hunk 2 above)
Comment (prprabhu-ms) [reply ui-library-add-call-recording/api/stopRecording.cs:13]: Thanks for that reference. This is the template generated by vscode. We do want the API endpoint name to be camelCase. I'll rename the classes in the next PR.

File: ui-library-add-call-recording/api/types.cs (hunk 3 above)
Comment (prprabhu-ms) [reply ui-library-add-call-recording/api/types.cs:5]: Thanks for pointing that out. This particular type is gone in the very next PR, so letting it be.

File: ui-library-add-call-recording/api/startRecording.cs (hunk 4 above)
Comment (prprabhu-ms) [reply ui-library-add-call-recording/api/startRecording.cs:20]: Checking as I implement the start/stop functions for real.

**Pull request merged**